src/averageAges.js

```diff
@@ -15,13 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  let male = people.filter(person => person.sex === 'm');
+
+  male = century
+    ? male.filter(person => Math.ceil(person.died / 100) === century)
+    : male;
+
+  const sumAvgAges = male.reduce(
+    (sum, person) => sum + (person.died - person.born), 0);
+  const avgAges = sumAvgAges / male.length;
```

Comment:
Move to the separate function & reuse it
```suggestion
  const sumAvgAges = getAvarageAge(male);
```